src/app/pages/map-view/sidebar-section/sidebar-section.component.ts

```diff
@@ -0,0 +1,250 @@
+        if (type === 'poly') {
+            (pointDrawer as L.Handler).disable();
+            (polygonDrawer as L.Handler).enable();
+
+        } else {
+            (polygonDrawer as L.Handler).disable();
+            (pointDrawer as L.Handler).enable();
+        }
+    }
+
+    onOpacityChange(name: string, opacity: number): void {
+        const el = this.filterByName(name);
+        const lyr = this.layersMap.get(name);
+        (lyr as L.TileLayer).setOpacity(el.opacity);
+        this.layersMap.set(name, lyr);
+        this.layers = Array.from(this.layersMap.values());
+    }
+
+    onShowChange(name: string, show: boolean): void {
+        const visible = show ? 'visible' : 'hidden';
+        const lyr = this.layersMap.get(name);
+        const el = this.filterByName(name);
+        if (show) {
+            this._layerService.getLayer(el).subscribe(res => {
+                (res as L.TileLayer).setOpacity(el.opacity);
+                this.layersMap.set(name, res);
+                this.layers = Array.from(this.layersMap.values());
+            });
+        } else {
+            this.layersMap.delete(name);
+            this.layers = Array.from(this.layersMap.values());
+        }
+        // all card panels should be closed;
+        el.expanded = undefined;
+    }
+
+    filterByName(name: string): LayerCard {
+        return this.cards.filter(el => {
+            if (el.info.name === name) {
+                return true;
+            }
+        })[0];
+    }
+    onPaletteChange(name: string, palette: string): void {
+        const el = this.filterByName(name);
+        this._layerService.getLayer(el).subscribe(res => {
+            res.setOpacity(el.opacity);
+            this.layersMap.set(el.info.name, res);
+            this.layers = Array.from(this.layersMap.values());
+        });
+    }
+
+    onValuesChange(name: string, layer: L.Layer): void {
+        const el = this.filterByName(name);
+        this._layerService.getLayer(el).subscribe(res => {
+            res.setOpacity(el.opacity);
+            this.layersMap.set(el.info.name, res);
+            this.layers = Array.from(this.layersMap.values());
+        });
+        if (el.hasOwnProperty('mask') && el.mask !== undefined) {
+            this.isLoading = true;
+            const zoom = this.map.getZoom();
+            let values = el.values;
+            if (el.info.name === 'change-detection') {
+                values = this.cards.filter(pt => pt.info.name === 'creation-render')[0].values;
+            }
+            this._layerService.getSummary(el, values, zoom).subscribe(res => {
+                el.summary = res;
+                el.expanded = 'summary';
+                this.isLoading = false;
+            }, console.error);
+        }
+    }
+
+    constructor(
+        private _layerService: LayerService,
+
+    ) {
+    }
+    ngOnInit() {
+        this.isSingle = (this.cards && this.cards.length === 1) ? true : false;
+        // for debugging
```

Comment:
Generally remove any code commented out intended for debugging. This code was probably only useful for you for the specific issue you were attempting to resolve. Others can add it as necessary during their workflow. If there's a systemic problem where  debug logging is useful in a particular location, it probably means a refactor is warranted to fix an issue or more heavy handed logging is necessary.